src/app/backend/handler/apihandler.go

```diff
@@ -220,22 +243,22 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 
 	apiV1Ws.Route(
 		apiV1Ws.POST("/appdeployment").
-			To(apiHandler.handleDeploy).
+			To(apiHandler.injectApiClient(apiHandler.handleDeploy)).
```

Comment:
Is it an idea to use the Filter chaining and storing the client in the Request.Attributes? the comment on the attributes member of the Request struct says it is for request scoped variables, and it seems like the client should be. Error handling can be done in the filter too (see the CSRF filter)
[https://github.com/emicklei/go-restful/blob/master/request.go#L18](https://github.com/emicklei/go-restful/blob/master/request.go#L18)